VincentCATILLON/react-native-confetti-cannon
# Android hardware artifacting fix

## Description

Making two tweaks to handle artifacting issues that are occurring on different Android devices.

### Changes

- [x] Add `{ perspective: 100 }` transform to the transform array (see documentation [here](https://reactnative.dev/docs/animations#bear-in-mind)).
- [x] Add new `renderToHardwareTextureAndroid` prop that defaults to `true` (see documentation [here](https://reactnative.dev/docs/view#rendertohardwaretextureandroid))
- [x] Update project readme to include documentation on new `renderToHardwareTextureAndroid` prop 

### Validation

**Please Note:** To verify the artifacting issue is resolved, you'll need an Android device such as a Pixel 4, a recent Samsung device, or any of the OnePlus devices.

0. Unlink the project from `npm` if already done, so you can confirm the presence of artifacting before testing
    ```bash
    > npm unlink
    ```
0. Install latest dependencies by running
    ```bash
    > yarn install
    > cd example && npm install
    ```
1. Launch the application with the current implementation, confirming you see artifacting, by running
    ```bash
    > npm run android
    ```
1. Copy over the latest project changes to the example project by running
    ```bash
    > npm run copy-files
    ```
1. Confirm metro reloads the example project, then test the different animations and confirm there's no more artifacting issues on Android

### Result

You should no longer have artifacting issues when running the application.  Example before and after below on a OnePlus 7 Pro.

<table>
    <thead>
        <tr>
            <th>Before</th>
            <th>After</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                <a href="https://user-images.githubusercontent.com/4258033/108603304-43c0aa80-7375-11eb-8297-dcf8abaac09e.mp4" rel="noreferrer noopener" target="_blank">Before Video</a>
            </td>
            <td>
                <a href="https://user-images.githubusercontent.com/4258033/108603314-50450300-7375-11eb-8d3d-b2ca477e6600.mp4" rel="noreferrer noopener" target="_blank">After Video</a>
            </td>
        </tr>
    </tbody>
</table>




### Related issues

- #30
- #36
